components/Products.tsx

```diff
@@ -42,7 +47,24 @@ const FETCH_STORE = gql`
   }
 `
 
-const NFT = ({ media, title }: { media: string; title: string }) => {
+const useBuy = (tokenID: string, tokenPrice: string) => {
+  const { wallet } = useWallet();
+  const tokenPriceNumber = Number(tokenPrice) ;
+  tokenPrice = (tokenPriceNumber).toLocaleString('fullwide', {useGrouping:false})
```

Comment:
Can you explain this line here?

Reply:
Line 52: Converts string into a number. But it is still represented as 1e+24.
Line 53: Converts 1e+24 to 1000000(24 times) and then convert it to a string again for make_offer to consume.

Reply:
That makes sense, but it's not conventional. Have you taken a look at https://near.github.io/near-api-js/modules/utils_format.html ?

Reply:
Right, I didn't think to explore any official NEAR way to do it. In the first look it seems I can use "parseNearAmount" function. I will try this and run some tests.